viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
+        movie_genre = movie["genre"]
+        
+        if movie_genre not in movie_genre_dict:
+            movie_genre_dict[movie_genre] = 1
+        else:
+            movie_genre_dict[movie_genre] += 1
+    
+    for genre, count in movie_genre_dict.items():
+        if count > max_count:
+            max_count = count
+            most_watched_genre = genre 
+    
+    return most_watched_genre
+    
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
-
+def get_unique_watched(user_data):
+    friends_watched_list = []    
+    unique_list_of_dicts = []
+    friends = user_data["friends"]
+    user_watched = user_data["watched"]
+        
+    for friend in friends:
+        for movie in friend["watched"]:
+            friends_watched_list.append(movie["title"])
+                
+    for movie in user_watched:
+        if movie["title"] not in friends_watched_list:
+            unique_list_of_dicts.append(movie)
         
+    return unique_list_of_dicts 
+
+
+def get_friends_unique_watched(user_data):
+    friends_watched_list = []
+    unique_friends_movies = []
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_watched_list:
+                friends_watched_list.append(movie)
+                
+    for movie in friends_watched_list:
+        if movie not in user_data["watched"]:
+            unique_friends_movies.append(movie)
+            
+    return unique_friends_movies
+            
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
-
+def get_available_recs(user_data):
+    user_subscriptions = [] 
+    recommended_movies = []
+    
+    # get user list of subscriptions
+    for sub in user_data["subscriptions"]:
+        user_subscriptions.append(sub)  
+        
+    # use unique friend's function, list of movie dicts
+    friends_unique = get_friends_unique_watched(user_data)
+    for movie in friends_unique:
+        if movie["host"] in user_subscriptions:
+            recommended_movies.append(movie)
+    
+    # return list of recommended movies 
+    return recommended_movies
+                    
+    
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+    fav_genre = get_most_watched_genre(user_data)
+    
+    # unique friend's function, returns list of movie dicts
+    friends_unique = get_friends_unique_watched(user_data)
+    for movie in friends_unique:
+        if movie["genre"] == fav_genre:
+            recommended_movies.append(movie)
+    
+    return recommended_movies 
+
+
+def get_rec_from_favorites(user_data):
+    recommended_movies = []
+    user_watched = get_unique_watched(user_data)

```

Comment:
Small naming nitpick: `user_watched` feels a bit misleading since this isn't all the movies the user watched, just the ones they've watched that friends have not. `user_unique_watched` would be an example of a less misleading name.